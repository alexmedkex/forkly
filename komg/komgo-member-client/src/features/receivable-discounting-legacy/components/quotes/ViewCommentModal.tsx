import React from 'react'
import { Modal, Button } from 'semantic-ui-react'
import styled from 'styled-components'
import { blueGrey } from '../../../../styles/colors'

export interface IViewCommentModalProps {
  open: boolean
  commentText: string
  date: string
  bankName: string
  handleClosed: () => void
}

const ViewCommentModal: React.FC<IViewCommentModalProps> = ({ commentText, date, bankName, handleClosed, open }) => (
  <Modal size="large" open={open}>
    <Modal.Header>Comment history</Modal.Header>
    <Modal.Content>
      <BankName>{bankName}</BankName>
      <Date>{date}</Date>
      <CommentBody>{commentText}</CommentBody>
    </Modal.Content>
    <Modal.Actions>
      <Button data-test-id="close-comments" onClick={handleClosed} content="Close" primary={true} />
    </Modal.Actions>
  </Modal>
)

const BankName = styled.b``
const Date = styled.p`
  color: ${blueGrey};
  font-size: 13px;
`
const CommentBody = styled.p``

export default ViewCommentModal
