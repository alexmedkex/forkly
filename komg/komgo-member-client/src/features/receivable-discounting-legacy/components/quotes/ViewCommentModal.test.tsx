import React from 'react'
import ViewCommentModal, { IViewCommentModalProps } from './ViewCommentModal'
import { shallow } from 'enzyme'
import { render } from '@testing-library/react'

describe('ViewCommentModal', () => {
  let defaultProps: IViewCommentModalProps
  beforeEach(() => {
    defaultProps = {
      open: true,
      commentText: 'This is a test comment',
      date: '2015/03/19',
      bankName: 'Mercuria',
      handleClosed: jest.fn()
    }
  })

  it('should match snapshot', () => {
    expect(render(<ViewCommentModal {...defaultProps} />).asFragment()).toMatchSnapshot()
  })

  it('should call handleClosed when closed is clicked', () => {
    const wrapper = shallow(<ViewCommentModal {...defaultProps} />)

    const close = wrapper.find('[data-test-id="close-comments"]')
    close.simulate('click')

    expect(defaultProps.handleClosed).toHaveBeenCalled()
  })
})
