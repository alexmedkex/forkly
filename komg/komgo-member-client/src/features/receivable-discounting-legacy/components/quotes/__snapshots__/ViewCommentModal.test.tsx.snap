// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewCommentModal should match snapshot 1`] = `<DocumentFragment />`;
